vilebot/src/main/java/com/oldterns/vilebot/handlers/user/News.java

```diff
@@ -0,0 +1,202 @@
+            newsFeedsByCategory.put( "saskatoon", new URL( "https://rss.cbc.ca/lineup/canada-saskatoon.xml" ) );
+            newsFeedsByCategory.put( "manitoba", new URL( "https://rss.cbc.ca/lineup/canada-manitoba.xml" ) );
+            newsFeedsByCategory.put( "thunderbay", new URL( "https://rss.cbc.ca/lineup/canada-thunderbay.xml" ) );
+            newsFeedsByCategory.put( "sudbury", new URL( "https://rss.cbc.ca/lineup/canada-sudbury.xml" ) );
+            newsFeedsByCategory.put( "windsor", new URL( "https://rss.cbc.ca/lineup/canada-windsor.xml" ) );
+            newsFeedsByCategory.put( "london", new URL( "https://www.cbc.ca/cmlink/rss-canada-london" ) );
+            newsFeedsByCategory.put( "waterloo", new URL( "https://rss.cbc.ca/lineup/canada-kitchenerwaterloo.xml" ) );
+            newsFeedsByCategory.put( "toronto", new URL( "https://rss.cbc.ca/lineup/canada-toronto.xml" ) );
+            newsFeedsByCategory.put( "hamilton", new URL( "https://rss.cbc.ca/lineup/canada-hamiltonnews.xml" ) );
+            newsFeedsByCategory.put( "montreal", new URL( "https://rss.cbc.ca/lineup/canada-montreal.xml" ) );
+            newsFeedsByCategory.put( "nb", new URL( "https://rss.cbc.ca/lineup/canada-newbrunswick.xml" ) );
+            newsFeedsByCategory.put( "pei", new URL( "https://rss.cbc.ca/lineup/canada-pei.xml" ) );
+            newsFeedsByCategory.put( "ns", new URL( "https://rss.cbc.ca/lineup/canada-novascotia.xml" ) );
+            newsFeedsByCategory.put( "newfoundland", new URL( "https://rss.cbc.ca/lineup/canada-newfoundland.xml" ) );
+            newsFeedsByCategory.put( "north", new URL( "https://rss.cbc.ca/lineup/canada-north.xml" ) );
+        }
+        catch ( MalformedURLException e )
+        {
+            logger.error( "Error loading news URLs" );
+            throw new RuntimeException( e );
+        }
+    }
+
+    private static final Pattern NEWS_PATTERN = Pattern.compile( "^!news(?: ([a-zA-Z]{2,})|)" );
+
+    private static final Pattern NEWS_HELP_PATTERN = Pattern.compile( "^!news help" );
+
+    private static final String helpMessage = generateHelpMessage();
+
+    @Override
+    public void onGenericMessage( final GenericMessageEvent event )
+    {
+        String text = event.getMessage();
+        Matcher matcher = NEWS_PATTERN.matcher( text );
+        Matcher helpMatcher = NEWS_HELP_PATTERN.matcher( text );
+
+        if ( helpMatcher.matches() )
+        {
+            event.respondPrivateMessage( helpMessage );
+        }
+        else if ( matcher.matches() )
+        {
+            currentNews( event, matcher );
+        }
+    }
+
+    private void currentNews( GenericMessageEvent event, Matcher matcher )
+    {
+        String category = matcher.group( 1 ); // The news category
+        if ( category == null )
+        {
+            category = defaultCategory;
+        }
+        category = category.toLowerCase();
```

Comment:
Currently this will perform an extra operation in the null case (it will lower case default category without needing to). Maybe use a ternary operator instead? i.e. category = (category != null) ? category.toLowerCase() : defaultCategory;